src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -134,9 +137,28 @@ class ResourceService(private val resourceTypes: Map[ResourceTypeName, ResourceT
     accessPolicyDAO.createPolicy(AccessPolicy(resourceAndPolicyName, members, email, roles, actions))
   }
 
-  def listUserAccessPolicies(resourceType: ResourceType, userInfo: UserInfo): Future[Set[ResourceIdAndPolicyName]] = {
-    accessPolicyDAO.listAccessPolicies(resourceType.name, userInfo.userId)
-  }
+  def listUserAccessPolicies(resourceTypeName: ResourceTypeName, userId: WorkbenchUserId): IO[Set[UserPolicyResponse]] = for{
+      rt <- IO.fromEither(resourceTypes.get(resourceTypeName).toRight(new WorkbenchException(s"missing configration for resourceType ${resourceTypeName}")))
+      isConstrained = rt.isAuthDomainConstrainable
+      ridAndPolcyName <- accessPolicyDAO.listAccessPolicies(resourceTypeName, userId) // List all policies of a given resourceType the user is a member of
+      rids = ridAndPolcyName.map(_.resourceId)
+      resources <- accessPolicyDAO.listResourceWithAuthdomains(resourceTypeName, rids)
+      authDomainMap = resources.map(x => x.resourceId -> x.authDomain).toMap
+      results = ridAndPolcyName.toList.map{
+        rnp =>
+          if(!isConstrained)
+            EitherT.fromEither[IO](UserPolicyResponse(rnp.resourceId, rnp.accessPolicyName, Set.empty, Set.empty).asRight[String])
+          else{
+            for{
+              authDomains <- EitherT.fromEither[IO](authDomainMap.get(rnp.resourceId).toRight(s"no auth domain found for ${rnp.resourceId}"))
```

Comment:
I am not certain, but it seems that this code treats a resource with no auth domain as an error. That is not true. The resource type may allow auth domains but that does not mean all resources of that type have them.

Reply:
The error message is a bit confusing.

`ridAndPolcyName <- accessPolicyDAO.listAccessPolicies(resourceTypeName, userId)`
```      resources <- if(isConstrained) accessPolicyDAO.listResourceWithAuthdomains(resourceTypeName, rids) else IO.pure(Set.empty)
      authDomainMap = resources.map(x => x.resourceId -> x.authDomain).toMap```

this line is saying if there's a resource that exists in `ridAndPolcyName`, but not in `authDomainMap` that's returned by the second query, we treat it as an error. Up for suggestions for the error message

Reply:
just a reminder to myself..log it